read_buffer/src/table.rs

```diff
@@ -126,7 +126,9 @@ impl Table {
 
     /// The total size of the table in bytes.
     pub fn size(&self) -> u64 {
-        self.table_data.read().unwrap().meta.size
+        let base_size = std::mem::size_of::<Self>() + self.name.len();
+        // meta.size accounts for all the row group data.
+        base_size as u64 + self.table_data.read().unwrap().meta.size()
```

Comment:
```suggestion
        base_size as u64 + self.table_data.expect("mutex poisoned").unwrap().meta.size()
```
This threw me for a loop for a moment -- "is it OK to unwrap the result of `read()` I was about to ask" lol

Reply:
@carols10cents is probably gonna get this when we add parking lot so I'll leave it out for now.

Reply:
avoiding the need to constantly check for a poisoned mutex will be reason enough for me to switch to a different locking library.  LOL